File: ApiIntegrador/Dto/SuscriptorDeudaDto.cs
namespace ApiIntegrador.Dto
{
 public class SuscriptorDeudaDTO
    {
        public int IdSuscriptor { get; set; }
        public string Nombre { get; set; } = string.Empty;
         public string Colonia { get; set; }      
        public string Ciudad { get; set; } 
        public decimal DeudaTotal { get; set; }
        public List<ContratoDeudaDetalleDTO> Contratos { get; set; } = new();
    }
}

File: gestor-pagos-frontend/src/app/modelos/SuscriptorDeudaDTO.ts
export interface PromocionDetalleDTO {
  nombre: string;
  descuentoAplicado: number;
  fechaInicio: string;
  fechaFin: string;
  diasDuracion: number;
}

export interface ContratoDeudaDetalleDTO {
  idContrato: number;
  paquete: string;
  tipoContrato: string;
  precioOriginal: number;
  descuentoTotal: number;
  totalAPagar: number;
  promociones: PromocionDetalleDTO[];
}

export interface SuscriptorDeudaDTO {
  idSuscriptor: number;
  nombre: string;
  colonia: string;
  ciudad: string;
  deudaTotal: number;
  contratos: ContratoDeudaDetalleDTO[];
}
